presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -1371,20 +1371,22 @@ else if (u21 >= 0) {
         return (int) qhat;
     }
 
-    private static long divideUnsignedLong(long dividend, int divisor)
+    /**
+     * Hacker's Delight 9-3
```

Comment:
Any referred implementation  of this algorithm ?

Reply:
I just copied it from the book =))